### PR TITLE
rpc, test: addpeeraddress test coverage, code simplify/constness

### PR DESCRIPTION
**Pull request by jonatack**
- Add functional test coverage for rpc addpeeraddress
- Simplify addpeeraddress and improve code constness

### CODE REVIEW

**Review by klementtan — APPROVED** [review]
ACK [`b36e0cd`](https://github.com/bitcoin/bitcoin/pull/22043/commits/b36e0cd1b9d361ac6f9777c09328a13e9ee923be)

Tested with the following scenarios:
**Success: peer added**
```bash
./src/bitcoin-cli -signet addpeeraddress 1.1.1.1 38333  
{
  "success": true
}

# Check if peer address added
./src/bitcoin-cli -signet getnodeaddresses 0    
[
...,
  {
    "time": 1621873900,
    "services": 9,
    "address": "1.1.1.1",
    "port": 38333,
    "network": "ipv4"
  },
...
]
```
**Fail: Peer address already added**
```bash
# Continue after previous scenario
./src/bitcoin-cli -signet addpeeraddress 1.1.1.1 38333
{
  "success": false
}

```

**Fail: Empty Address**
```bash
./src/bitcoin-cli -signet addpeeraddress "" 38333
{
  "success": false
}
```

**Comment by reemuru** [comment]
Tested ACK. Run on signet, Fedora 34 and Ubuntu 20.x

<details hidden><summary>rpc log</summary>
   
    [user@fedora bitcoin]$ ./src/bitcoin-cli -signet addpeeraddress x.xxx.xxx.x 38333
    {
      "success": true
    }
    [user@fedora bitcoin]$ ./src/bitcoin-cli -signet getnodeaddresses 0
    [
      {
       "time": 1621866922,
        "services": 1033,
        "address": "xxxx:xxxx:xxxx:x:xxxx:xxxx:xxxx:xxxxx",
        "port": 38333,
        "network": "ipv6"
      }
    ]
    [user@fedora bitcoin]$ ./src/bitcoin-cli -signet addpeeraddress SAMEADDRESS 38333
    {
      "success": false
    }
    [user@fedora bitcoin]$ ./src/bitcoin-cli -signet addpeeraddress "" 38333
    {
      "success": false
    }

</details> 

<details hidden><summary>test log</summary>

    [user@fedora bitcoin]$ ./test/functional/rpc_net.py 
    2021-05-25T03:37:33.624000Z TestFramework (INFO): Initializing test directory /tmp/bitcoin_func_test_ulr3n_ye
    2021-05-25T03:37:34.200000Z TestFramework (INFO): Test getconnectioncount
    2021-05-25T03:37:34.201000Z TestFramework (INFO): Test getpeerinfo
    2021-05-25T03:37:37.237000Z TestFramework (INFO): Test getnettotals
    2021-05-25T03:37:37.360000Z TestFramework (INFO): Test getnetworkinfo
    2021-05-25T03:37:37.511000Z TestFramework (INFO): Test getaddednodeinfo
    2021-05-25T03:37:37.528000Z TestFramework (INFO): Test service flags
    2021-05-25T03:37:37.697000Z TestFramework (INFO): Test getnodeaddresses
    2021-05-25T03:37:43.703000Z TestFramework (INFO): Test addpeeraddress
    2021-05-25T03:37:43.766000Z TestFramework (INFO): Stopping nodes
    2021-05-25T03:37:43.920000Z TestFramework (INFO): Cleaning up /tmp/bitcoin_func_test_ulr3n_ye on exit
    2021-05-25T03:37:43.921000Z TestFramework (INFO): Tests successful

</details>

**Review by reemuru — APPROVED** [review]

**Review by maflcko — COMMENTED** [review]
review ACK b36e0cd1b9d361ac6f9777c09328a13e9ee923be 💭

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

review ACK b36e0cd1b9d361ac6f9777c09328a13e9ee923be 💭
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUjPyQv/bnekUdWEbiGDKF6KIJPFafnLZSiXouANoRjNsyp61MhZjX/VPfCkAfap
bxECMzdVChRA4qS/oDKN7v6PTw0QkizRuOBRpzMmLKzGVG+sHJN/rYRXeQIQTdc6
fu2rAvjEz5ab85bzTjhY+yZ1pgMQWfbhXv44vT//m9lv7V3UytAZ86/z0SQqtBaT
5AG9p+JfY4S3oSbxp7RY9qsw8XBPAt/73xDgnWn25f+9IqxGgwSzAuGFniHyg7WG
hwzBIYxvG8oYp6a7fUx6wqahO4DA5YVRdDrTNQSuIr+11s8iP5i6E7V0E/7fxP4S
W2WEH008QtFkGqUYHmQOElSLn4JrlgCyAc8CkuLnctBny6E/GpJqqFuoSDsMba5A
Mp0ynrMFP2mYbwIShwC88WMwc+H742Z92Z4ywhHihuKsxG2p87CoSDCjhWse0jR9
Yq9t6cvuopmRPJncMX88SN0L99hZ5sHrktwMSg7Uynhq/Us7j57S7yIyCH+Prg8O
dvsOmcoH
=UOZH
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `9dce392a47e73871b02708975647ca954f17d4677bc589be4b34303e8fca8153  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e8929401089dce392a47e73871b02708975647ca954f17d4677bc589be4b34303e8fca8153f01082d6c4b911c25e40b1971629e01003aa08fff010261239916a40069434c21c731a343b1908f020df22b31bccd8af24840cba73a67900dfae98f69fae37d5811e3afbbdf61106ff08f10460aca9b7f008a2be758fbe573b480083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff0104a1e11c19b49ffc00c6a488ed421a7a808f0206b1543c9ca76eced42d2ff20b9aa47eb5ce01c4e6a26ca39b93651c85f7e2eb408f02052693a460d4e2976e83738b523fd0192368f57277e4beffe683598877bb9f17f08f020b57fb9ad9c780e5ca6d1b9943d0c2ab22b9a61e97359d72ed080f597088ba9ae08f10460aca9b8f008de34f70f002bd0460083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267f010c176e01e71af2b54b3a7b61c04cf530c08f10460aca9b9f0081255edf2a2153bd00083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6d)

</details>

File: src/rpc/net.cpp [hunk 1]
```diff
@@ -931,26 +931,22 @@ static RPCHelpMan addpeeraddress()
...
-    address.nTime = GetAdjustedTime();
-    // The source address is set equal to the address. This is equivalent to the peer
-    // announcing itself.
-    if (!node.addrman->Add(address, address)) {
-        obj.pushKV("success", false);
-        return obj;
+    bool success{false};
+
+    if (LookupHost(addr_string, net_addr, false)) {
+        CAddress address{CAddress({net_addr, port}, ServiceFlags(NODE_NETWORK | NODE_WITNESS))};
```
Comment (MarcoFalke) [inline src/rpc/net.cpp:942]: could remove duplicate `CAddress` constructor

```suggestion
        CAddress address{{net_addr, port}, ServiceFlags{NODE_NETWORK | NODE_WITNESS}};
```

**Pull request merged**